dvl-a50/dvl.py

```diff
@@ -175,12 +194,44 @@ def set_should_send(self, should_send):
         self.should_send = should_send
         self.save_settings()
 
+    def longitude_scale(self, lat):
+        scale = math.cos(lat * (math.pi / 180.0))
+        return max(scale, 0.01)
+
+    def latLngToXY (self, lat, lon):
+        x = (lat-self.origin[0]) * LATLON_TO_CM
+        y =  self.longitude_scale((lat + self.origin[0])/2) * LATLON_TO_CM * (lon-self.origin[1])
+        return [x, y]
+
+    def has_origin_set(self):
+        data = json.loads(self.mav.get("/GLOBAL_POSITION_INT/message"))
+        return data['lat'] != 0 or data['lon'] != 0
+
+    def set_current_position(self, lat, lon):
+        """
+        Sets the EKF origin to lat, lon
+        """
+        lat = float(lat)
+        lon = float(lon)
+        if not self.has_origin_set():
+            self.mav.set_gps_origin(lat, lon)
+            self.origin = [lat, lon]
+            self.save_settings()
```

Comment:
Should this be `self.set_gps_origin(lat, lon)` instead? Otherwise why does that function exist?

Reply:
good catch, fixing